src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/ContactIdentityController.kt

```
@@ -0,0 +1,119 @@
+      ),
+    ],
+  )
+  @PreAuthorize("hasAnyRole('ROLE_CONTACTS_ADMIN')")
+  fun create(
+    @PathVariable("contactId") @Parameter(
+      name = "contactId",
+      description = "The id of the contact",
+      example = "123456",
+    ) contactId: Long,
+    @Valid @RequestBody request: CreateIdentityRequest,
+  ): ResponseEntity<Any> {
+    val created = contactIdentityFacade.create(contactId, request)
+    return ResponseEntity
+      .status(HttpStatus.CREATED)
+      .body(created)
+  }
+
+  @GetMapping("/{contactIdentityId}")
+  @Operation(
+    summary = "Get an identity",
+    description = "Gets a contacts identity by id",
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Found the identity successfully",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ContactIdentityDetails::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Could not find the the contact or identity this request is for",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @PreAuthorize("hasAnyRole('ROLE_CONTACTS_ADMIN')")
+  fun get(
+    @PathVariable("contactId") @Parameter(
+      name = "contactId",
+      description = "The id of the contact",
+      example = "123456",
+    ) contactId: Long,
+    @PathVariable("contactIdentityId") @Parameter(
+      name = "contactIdentityId",
+      description = "The id of the contact identity",
+      example = "987654",
+    ) contactIdentityId: Long,
+  ): ResponseEntity<Any> {
+    return contactIdentityFacade.get(contactId, contactIdentityId)
+      ?.let { ResponseEntity.ok(it) }
+      ?: throw EntityNotFoundException("Contact identity with id ($contactIdentityId) not found for contact ($contactId)")
```

can we throw this exception in the service layer instead of the controller, 

I know it's bit inconsistent at the moment, some have done it in the repository layer, 
I think the best place should be in the service layer.

---

I think i'll move it into the facade, it's nice the service can return an optional result for other use cases but take your point maybe controller not the best. I actually hate throw exceptions for this stuff and would rather handle with a return type but that's a topic for another day :)